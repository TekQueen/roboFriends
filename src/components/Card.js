import React from 'react';

const Card = ({name, email, id, username}) => 
    <div className = 'tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5'>
        <img src = {`https://robohash.org/${id}?size=200x200`} alt = 'robot' />
        <div className = 'robotInfoContainer'>
            <h2>{name}</h2>
            <h3>{username}</h3>
            <h3>{email}</h3>
        </div>
    </div>

export default Card;