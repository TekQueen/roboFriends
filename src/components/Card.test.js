import { shallow, mount, render} from 'enzyme';
import React from 'react';
import Card from './Card';


describe('<Card />', () => {
  const CardComponent = shallow(<Card />);
  

  it('expect to render Card component', () => {
    expect.assertions(1);
    expect(CardComponent.length).toBe(1);
  })

  it('finds the node which matches the selecter', () => {
    expect.assertions(1);
    expect(CardComponent.find('.tc').length).toBe(1);
  })

  it ('returns a new wrapper with only host nodes', () => {
      expect.assertions(1);
      expect(CardComponent.find('.tc').hostNodes().length).toBe(1);
  })

  it('jest snapshot testing', () => {
      expect(CardComponent).toMatchSnapshot();
  })
})


