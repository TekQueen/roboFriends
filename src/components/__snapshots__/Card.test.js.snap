// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> jest snapshot testing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="robot"
          src="https://robohash.org/undefined?size=200x200"
/>,
        <div
          className="robotInfoContainer"
>
          <h2 />
          <h3 />
          <h3 />
</div>,
      ],
      "className": "tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "robot",
          "src": "https://robohash.org/undefined?size=200x200",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2 />,
            <h3 />,
            <h3 />,
          ],
          "className": "robotInfoContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="robot"
            src="https://robohash.org/undefined?size=200x200"
/>,
          <div
            className="robotInfoContainer"
>
            <h2 />
            <h3 />
            <h3 />
</div>,
        ],
        "className": "tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "robot",
            "src": "https://robohash.org/undefined?size=200x200",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2 />,
              <h3 />,
              <h3 />,
            ],
            "className": "robotInfoContainer",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
